### PR TITLE
audit: Don't mutate args

### PR DESCRIPTION
**Pull request by GauthamGoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Fixes bug https://github.com/Homebrew/brew/pull/5204#issuecomment-434239055
cc @commitay

### CODE REVIEW

File: Library/Homebrew/dev-cmd/audit.rb [hunk 1]
```diff
@@ -151,7 +150,12 @@ def audit
 
     new_formula_problem_lines = []
     ff.sort.each do |f|
-      options = { new_formula: new_formula, strict: strict, online: online, only: args.only, except: args.except }
+      if only_cops
```
Comment (reitermarkus) [inline Library/Homebrew/dev-cmd/audit.rb]:
```ruby
only = only_cops ? ["style"] : args.only
```

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/audit.rb]: I haven't seen ternary operator being used in the codebase so I thought its against our style :-|

Comment (scpeters) [reply Library/Homebrew/dev-cmd/audit.rb]: I see it in use a fair bit:

* https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L240
* https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L245
* https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L970

Comment (GauthamGoli) [reply Library/Homebrew/dev-cmd/audit.rb]: Cool. Updating the code.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/audit.rb]: I think it's personal preference at this point if it's simple; personally I find the ternary operator always harder to read but this is fine either way.

**Comment by MikeMcQuaid** [comment]
Thanks again @GauthamGoli!

**Pull request merged**